core/src/main/java/org/apache/accumulo/core/fate/accumulo/FateMutatorImpl.java

```diff
@@ -105,13 +108,20 @@ public FateMutator<T> putTxInfo(TxInfo txInfo, byte[] data) {
       case RETURN_VALUE:
         putReturnValue(data);
         break;
+      default:
+        throw new IllegalArgumentException("Unexpected TxInfo type " + txInfo);
     }
     return this;
   }
 
   @Override
   public FateMutator<T> putRepo(int position, Repo<T> repo) {
-    mutation.put(RepoColumnFamily.NAME, invertRepo(position), new Value(serialize(repo)));
+    final Text cq = invertRepo(position);
+    // ensure this repo is not already set
+    mutation.addCondition(new Condition(RepoColumnFamily.NAME, cq));
```

Comment:
We could add a stronger check that ensure the current max repo in storage is `position - 1`.  This could be done with a specialized iterator.  Or could have to conditions like :

 * A condition that checks position is absent
 * A condition that checks that position -1 is present.

Reply:
Using conditions, is there a way to verify that a column is present without knowing its value? To me it looks like we can only verify that a columns exists and verify that a column has a specific value, not that a column exists with any value.

Reply:
Made a note about this in the ticket: https://github.com/apache/accumulo/issues/4114#issuecomment-1900971079. Going to try to merge in what I have as is and handle that in another PR.